### ISSUE TITLE
Investigate unawaited calls in library code

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/test#972]
See calls to `unawaited` in https://github.com/dart-lang/test/commit/8a90f38080f4d313f9d87e633e2a5a9b36b5a1bd

Need to investigate.

CC @natebosch @grouma

**Comment by jakemac53** [comment]
I glanced through these and I don't see anything concerning, its mostly just cleanup code (deleting temp dirs after browsers shut down, closing streams on shutdown, etc).

I will leave it open for now but if somebody else takes a look and agrees then I think we can close this.

**Comment by jakemac53** [comment]
That would only be for debugging though right? Not awaiting them shouldn't have any negative impact.

**Comment by natebosch** [comment]
Yeah, the await would be intended to give us confidence in what we're looking at when debugging, I don't expect it should cause the hang.

**Comment by grouma** [comment]
I actually went through and addressed all unawaited futures a while back when I was investigating the Travis issue. Unfortunately it didn't help and that's why I abandoned the PR. I still think we should clean these up though so let's leave this open.
